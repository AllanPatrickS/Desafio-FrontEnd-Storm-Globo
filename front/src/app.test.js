import { cleanup } from '@testing-library/react';
import { create } from "react-test-renderer";
import App from './app';

describe('Testing app', () => {

  it('renders correctly', () => {
    const wrapper = create(<App />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  afterEach(function () {
    cleanup();
  });
});