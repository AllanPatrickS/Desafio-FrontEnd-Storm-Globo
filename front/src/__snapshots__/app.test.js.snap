// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing app renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-bar-1 mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <svg
        alt="brand"
      >
        symbol.svg
      </svg>
      <hr
        className="MuiDivider-root makeStyles-divider-2 MuiDivider-flexItem MuiDivider-vertical"
      />
      <div
        className="MuiButtonGroup-root makeStyles-type-3"
        role="group"
      >
        <button
          aria-label="type"
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined makeStyles-gradientType-4"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-white-6"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm0 10.99h7c-.53 4.12-3.28 7.79-7 8.94V12H5V6.3l7-3.11v8.8z"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="type"
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined makeStyles-whiteType-5"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-gray-11"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="MuiBox-root MuiBox-root-15 makeStyles-search-7"
      >
        <div
          className="MuiInputBase-root makeStyles-lightGray-9"
          onClick={[Function]}
        >
          <input
            aria-label="search"
            className="MuiInputBase-input makeStyles-lightGrayItalic-10"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Pesquisar…"
            type="text"
            value=""
          />
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-searchIconBox-8"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-gray-11"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </span>
        </button>
      </div>
      <hr
        className="MuiDivider-root makeStyles-space-12 MuiDivider-flexItem MuiDivider-vertical"
      />
      <div
        className="MuiBox-root MuiBox-root-16"
      >
        <button
          aria-label="filter"
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-whiteType-5"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-gray-11"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17v2h6v-2H3zM3 5v2h10V5H3zm10 16v-2h8v-2h-8v-2h-2v6h2zM7 9v2H3v2h4v2h2V9H7zm14 4v-2H11v2h10zm-6-4h2V7h4V5h-4V3h-2v6z"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="add"
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-add-13"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-white-6"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
            INCLUIR USUÁRIO
          </span>
        </button>
      </div>
      <hr
        className="MuiDivider-root makeStyles-divider-2 MuiDivider-flexItem MuiDivider-vertical"
      />
      <div
        className="MuiButtonGroup-root makeStyles-changes-14"
        role="group"
      >
        <button
          aria-label="home"
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-gray-11"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="settings"
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-gray-11"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
                transform="scale(1.2, 1.2)"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="power"
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-gray-11"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M13 3h-2v10h2V3zm4.83 2.17l-1.42 1.42C17.99 7.86 19 9.81 19 12c0 3.87-3.13 7-7 7s-7-3.13-7-7c0-2.19 1.01-4.14 2.58-5.42L6.17 5.17C4.23 6.82 3 9.26 3 12c0 4.97 4.03 9 9 9s9-4.03 9-9c0-2.74-1.23-5.18-3.17-6.83z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </header>,
  <div
    className="MuiBox-root MuiBox-root-49 makeStyles-body-38"
  >
    <div
      className="MuiContainer-root makeStyles-container-39"
    >
      <div
        className="MuiPaper-root MuiTableContainer-root makeStyles-tableContainer-40 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <table
          aria-label="user table"
          className="MuiTable-root"
          role={null}
        >
          <thead
            className="MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head makeStyles-headCell-42"
                scope="col"
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-hiddenButton-45"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "borderRadius": 4,
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiButton-label"
                  >
                    Deletar
                  </span>
                </button>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head makeStyles-headCell-42"
                scope="col"
              >
                USUÁRIO
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head makeStyles-headCell-42"
                scope="col"
              >
                EMAIL
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head makeStyles-headCell-42"
                scope="col"
              >
                DATA DE INCLUSÃO
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head makeStyles-headCell-42"
                scope="col"
              >
                DATA DE ALTERAÇÃO
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head makeStyles-headCell-42"
                scope="col"
              >
                REGRAS
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head makeStyles-headCell-42"
                scope="col"
              >
                STATUS
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head makeStyles-headCell-42"
                scope="col"
              >
                AÇÕES
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root"
            role={null}
          />
        </table>
      </div>
      <div
        aria-label="pagination"
        className="MuiButtonGroup-root makeStyles-pagination-43"
        role="group"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined makeStyles-paginationButton-46 Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "borderRadius": 4,
            }
          }
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Primeiro
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined makeStyles-paginationButton-46 Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "borderRadius": 4,
            }
          }
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Anterior
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined makeStyles-page-47"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "borderRadius": 4,
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            0
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined makeStyles-paginationButton-46 Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "borderRadius": 4,
            }
          }
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Próximo
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined makeStyles-lastButton-48 Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "borderRadius": 4,
            }
          }
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            último
          </span>
        </button>
      </div>
    </div>
  </div>,
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-bar-57 mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-orange-58"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <p
        className="MuiTypography-root makeStyles-orangeText-59 MuiTypography-body1"
      >
        NO AR
      </p>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-tv-60"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M21 3H3c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h5v2h8v-2h5c1.1 0 1.99-.9 1.99-2L23 5c0-1.1-.9-2-2-2zm0 14H3V5h18v12z"
        />
      </svg>
      <p
        className="MuiTypography-root makeStyles-tvText-61 MuiTypography-body1"
      >
        ENCONTRO - 
        
      </p>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-update-62"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M21 10.12h-6.78l2.74-2.82c-2.73-2.7-7.15-2.8-9.88-.1-2.73 2.71-2.73 7.08 0 9.79s7.15 2.71 9.88 0C18.32 15.65 19 14.08 19 12.1h2c0 1.98-.88 4.55-2.64 6.29-3.51 3.48-9.21 3.48-12.72 0-3.5-3.47-3.53-9.11-.02-12.58s9.14-3.47 12.65 0L21 3v7.12zM12.5 8v4.25l3.5 2.08-.72 1.21L11 13V8h1.5z"
        />
      </svg>
      <p
        className="MuiTypography-root makeStyles-updateText-63 MuiTypography-body1"
      >
        Última atualização em 
        <strong>
          
        </strong>
      </p>
      <p
        className="MuiTypography-root makeStyles-date-64 MuiTypography-body1"
      >
        <strong>
          6
        </strong>
         de 
        <strong>
          Novembro
        </strong>
         de 
        <strong>
          2020
        </strong>
      </p>
      <h4
        className="MuiTypography-root makeStyles-clock-65 MuiTypography-h4"
      >
        <strong>
          ⁄⁄ 
          12
          :
          17
          :
          56
        </strong>
      </h4>
    </div>
  </header>,
]
`;
